client/lib/deterministic-stringify/index.js

```diff
@@ -1,32 +1,39 @@
-function deterministStringify( source ) {
-	var namespace = [],
-		keys;
+export function deterministicStringify( source ) {
+	if ( source === null ) {
+		return 'null';
+	}
+
+	// Handle primitive data types:
+	// boolean, number, string, undefined
 	if ( typeof source !== 'object' ) {
 		if ( typeof source === 'undefined' ) {
 			return 'undefined';
 		}
+
 		if ( typeof source === 'string' ) {
-			return "'" + source.toString() + "'";
+			return `'${ source.toString() }'`;
 		}
+
 		return source.toString();
 	}
+
+	// arrays are objects too so we have to
+	// manually ask if this is _also_ an array
 	if ( Array.isArray( source ) ) {
-		source = source.sort();
-		source.forEach( function( item ) {
-			namespace.push( deterministStringify( item ) );
-		} );
-		return namespace.join( ',' );
-	}
-	if ( source === null ) {
-		return 'null';
+		return source
+			.slice() // sort mutates the original array!
+			.sort()
```

Comment:
@blowery from what I gathered it's because we're mostly using this as a reference or key and not a value, otherwise we'd use `JSON.stringify()`

I'm not sure if we'd break anything but changing this but if we were using it for _set_ membership then we could falsely identify two sets as different when they are the same.

Not really trying to _change_ the behavior in this PR but rather just refectory the code (and maybe fix a lurking bug if it's there)